### ISSUE TITLE
Matshow/imshow changes layout in matplotlib >= 3.3

### ISSUE THREAD

**Issue by matthias-k** [lukelbd/proplot#235]
### Description

I set up a subplots layout the way I need it for my figure and the subplots look good. But as soon as I start filling the axes with image-data using `matshow` or `imshow`, the layout is completely changed. It appears, the axes change their aspect ratio to one, resulting in large margins in the final plot.
### Steps to reproduce

```python
import proplot as plot
import numpy as np

f, axes = plot.subplots(nrows=2, ncols=5, aspect=4/3, wspace='3mm', hspace='3mm')
axes[0].matshow(np.random.randn(3,4))
```

**Expected behavior**: I expect to see a nicely aligned grid of axes with the given aspect ratio, the first axes should be completely filled with the random data, roughly like this (except for the missing data):
![Screenshot_20201115_001856](https://user-images.githubusercontent.com/4093707/99158907-304bd680-26d8-11eb-9f6b-be005c6f054a.png)

**Actual behavior**: The matshow command changes the layout:
![Screenshot_20201115_002011](https://user-images.githubusercontent.com/4093707/99158935-570a0d00-26d8-11eb-9cc1-0ed852de682b.png)

### Equivalent steps in matplotlib

Not sure whether this applies here, but with the matplotlib `subplots`, the layout is not changed after the first `matshow`.

```python
import matplotlib.pyplot as plt
import numpy as np

f, axes = plt.subplots(2, 5, figsize=(4*5, 3*2))
axes[0, 0].matshow(np.random.randn(3,4))
f.subplots_adjust(hspace=0.1, wspace=0.1, left=0.02, right=0.98, top=0.98, bottom=0.02)
```
![Screenshot_20201115_002427](https://user-images.githubusercontent.com/4093707/99158992-f4fdd780-26d8-11eb-8be8-f1992ae3b5ba.png)

### Proplot version

0.6.4

**Comment by matthias-k** [comment]
I noticed that the issues seems to be related to the reference axes. Everything is fine if I apply imshow to the second instead of the first axes:
![Screenshot_20201116_132441](https://user-images.githubusercontent.com/4093707/99252396-2b3d7300-280f-11eb-8310-4c35fa4977cb.png)

**Comment by matthias-k** [comment]
It appears this is a regression, since the same issue happens to examples from the documentation,e.g. the example from https://proplot.readthedocs.io/en/latest/subplots.html.

![image](https://user-images.githubusercontent.com/4093707/99253090-68eecb80-2810-11eb-9d3f-b3607690412b.png)

According to the documentation, it should look like
![image](https://user-images.githubusercontent.com/4093707/99253182-9b002d80-2810-11eb-8b34-fe74ae46673c.png)

**Comment by matthias-k** [comment]
I ran a `git bisect` binary search and found that the bug was apparently introduced in e7f76fa436a62f6c2c920982b3771a0ad8e6e270. That commit seems to change how different aspect settings for axes are handled but I don't know the code well enough to spot the actual problem. @lukelbd maybe you have an idea?

**Comment by pratiman-91** [comment]
@matthias-k 
I think what you are looking for is the change in the aspect ratio. If you want to set the aspect ration to be same as matplotlib then you have to set it to `axes.set_aspect('auto')`. For example,
```python
import proplot as plot
import numpy as np

f, axes = plot.subplots(nrows=2, ncols=5, aspect=4/3, wspace='3mm', hspace='3mm')
axes[0].matshow(np.random.randn(3,4))
axes.set_aspect('auto')
```
will yield
![test](https://user-images.githubusercontent.com/31694629/103287327-41427400-4a08-11eb-9eef-7536eb71b465.png)

**Comment by lukelbd** [comment]
Thanks for the report and your investigation. After some playing around it looks like e7f76fa did not introduce this bug -- everything works fine on master in matplotlib 3.2.2. This is another matplotlib 3.3+ bug, possibly related to #210

**Comment by lukelbd** [comment]
Eating my words -- it was e7f76fa (I have no idea what `git bisect` is but evidently I should learn). In matplotlib 3.3+ they translate `aspect='equal'` to `aspect=1` (these are synonyms since here "1" is in data units) but I confused matplotlib's convention of data-units "aspect" with proplot's convention of physical-units "aspect". So this otherwise innocuous change caused the whole algorithm to fail.

Now fixed by:  124ea3f

**Comment by matthias-k** [comment]
@lukelbd Thanks for fixing this! For the sake of completeness, `git bisect` is a git feature for efficiently finding the commit that introduced a certain bug. It can save a lot of time :). See https://git-scm.com/docs/git-bisect for a nice introduction
